components/ai_chat/core/browser/engine/engine_consumer_oai.cc

```diff
@@ -0,0 +1,234 @@
+}
+
+EngineConsumerOAIRemote::~EngineConsumerOAIRemote() = default;
+
+void EngineConsumerOAIRemote::ClearAllQueries() {
+  api_->ClearAllQueries();
+}
+
+bool EngineConsumerOAIRemote::SupportsDeltaTextResponses() const {
+  return true;
+}
+
+void EngineConsumerOAIRemote::GenerateRewriteSuggestion(
+    std::string text,
+    const std::string& question,
+    GenerationDataCallback received_callback,
+    GenerationCompletedCallback completed_callback) {
+  base::Value::List messages;
+
+  {
+    base::Value::Dict message;
+    message.Set("role", "user");
+    message.Set("content", base::StrCat({text, question}));
+    messages.Append(std::move(message));
+  }
+
+  {
+    base::Value::Dict message;
+    message.Set("role", "system");
+    message.Set("content",
+                "This is an excerpt user selected to be rewritten. Put your "
+                "rewritten of the excerpt in <response></response> tags.");
+    messages.Append(std::move(message));
+  }
+
+  api_->PerformRequest(model_, std::move(messages),
+                       std::move(received_callback),
+                       std::move(completed_callback));
+}
+
+void EngineConsumerOAIRemote::GenerateQuestionSuggestions(
+    const bool& is_video,
+    const std::string& page_content,
+    SuggestedQuestionsCallback callback) {
+  const std::string& truncated_page_content =
+      page_content.substr(0, max_page_content_length_);
+  std::string content_segment = base::ReplaceStringPlaceholders(
+      l10n_util::GetStringUTF8(is_video
+                                   ? IDS_AI_CHAT_CLAUDE_VIDEO_PROMPT_SEGMENT
+                                   : IDS_AI_CHAT_CLAUDE_ARTICLE_PROMPT_SEGMENT),
+      {truncated_page_content}, nullptr);
+
+  base::Value::List messages;
+
+  {
+    base::Value::Dict message;
+    message.Set("role", "user");
+    message.Set("content", base::StrCat({content_segment, kQuestionPrompt}));
+    messages.Append(std::move(message));
+  }
+
+  {
+    base::Value::Dict message;
+    message.Set("role", "system");
+    message.Set("content",
+                "Here are three questions the user may ask about the content in <question> tags:\n");
+    messages.Append(std::move(message));
+  }
+
+  api_->PerformRequest(
+      model_, std::move(messages), base::NullCallback(),
+      base::BindOnce(
+          &EngineConsumerOAIRemote::OnGenerateQuestionSuggestionsResponse,
+          weak_ptr_factory_.GetWeakPtr(), std::move(callback)));
+}
+
+void EngineConsumerOAIRemote::OnGenerateQuestionSuggestionsResponse(
+    SuggestedQuestionsCallback callback,
+    GenerationResult result) {
+  if (!result.has_value() || result->empty()) {
+    // Query resulted in error
+    LOG(ERROR) << "Error getting question suggestions.";
+    std::move(callback).Run(base::unexpected(std::move(result.error())));
+    return;
+  }
+
+  std::vector<std::string> questions =
+      base::SplitString(*result, "|", base::WhitespaceHandling::TRIM_WHITESPACE,
```

Comment:
Reminder to update this parsing function to extract from `<question>Q</question>`

Reply:
added a parser